### ISSUE TITLE
yaxs='r' breaks smart limits

### ISSUE THREAD

**Issue by lindsayplatt** [USGS-R/gsplot#203]
yaxs='r' adds a buffer to the edge of the plot, but if a user specifies ylim=0, it adds a buffer and does not start exactly at 0. 

``` r
gs <- gsplot() %>%
    points(1:3,2:4,ylim=c(0,NA)) %>%
    error_bar(1,2, y.high=0.5, y.low=0.25)
gs
```

Possibly create a condition where a user-specified ylim forces yaxs='i' instead of 'r'?

**Comment by ldecicco-USGS** [comment]
Not specific to 0 (although I think that's a common use case). This...once again....goes to our general philosophy...if a user does confusing things (specifies limits but also asks for an auto buffer)...what do we want to happen?

**Comment by lindsayplatt** [comment]
My suggestion is to use yaxs/xaxs = "i" if a user gives ylim/xlim values. Otherwise, keep yaxs/xaxs="r". 

In `calc_views`, add `ifelse(any(is.na(usr.lim)==FALSE), views$par[[paste0(var, "axs")]] <- "i", NA)` after usr.lim is defined in `set_view_lim`.

This would yield the following: 

``` r
gs <- gsplot() %>%
    points(1:3,2:4) %>%
    error_bar(1,2, y.high=0.5, y.low=0.25)
gs
```

## ![image](https://cloud.githubusercontent.com/assets/13220910/9314971/e5173b46-44f1-11e5-9e04-5ec834ec6f94.png)

``` r
gs <- gsplot() %>%
    points(1:3,2:4, xlim=c(0,4)) %>%
    error_bar(1,2, y.high=0.5, y.low=0.25)
gs
```

## ![image](https://cloud.githubusercontent.com/assets/13220910/9314940/b9236d52-44f1-11e5-90dc-8502aed102b2.png)

``` r
gs <- gsplot() %>%
    points(1:3,2:4, ylim=c(0,4)) %>%
    error_bar(1,2, y.high=0.5, y.low=0.25)
gs
```

![image](https://cloud.githubusercontent.com/assets/13220910/9314955/d17b45f0-44f1-11e5-9b60-bf917bf97cd8.png)

**Comment by lindsayplatt** [comment]
this will be addressed by #211
